src/spaces/undirected_graph.rs

```diff
@@ -158,24 +209,24 @@ mod tests {
         assert_eq!(graph.distance(1, 5), None);
     }
 
-    #[test]
-    fn restriction() {
-        let graph = create_graph();
-        let mut section = HashMap::new();
-        section.insert(1, Rc::new(1) as Rc<dyn Any>);
-        section.insert(2, Rc::new(2) as Rc<dyn Any>);
-        section.insert(3, Rc::new(3) as Rc<dyn Any>);
-
-        let set_to = vec![1, 2].into_iter().collect::<HashSet<_>>();
-        let restricted_section = graph.restriction(&set_to, &section);
-        println!("{:?}", restricted_section);
-        println!(
-            "{:?}",
-            restricted_section
-                .get(&1)
-                .unwrap()
-                .downcast_ref::<i32>()
-                .unwrap()
-        );
-    }
```

Comment:
im sure we can. im gonna try and resolve this before closing this PR but the main issue was my enforcement of `Eq`, and `Any` being... any lol. we can def get it to work if we narrowed the scope a bit down from `dyn Any`

Reply:
Since all of this is discrete and you'll have stalk/germs over single cells, you could make those traits while the structs are generic over a type. You could enforce `Eq` on them, e.g., `Data<T: Eq>` and then `impl <T> Stalk for Data<T: Eq>` or something. Just throwing it out there.